### PR TITLE
Doc + update CMSCategory

### PR DESCRIPTION
**Pull request by firstred**
| Questions | Answers |
| --- | --- |
| Branch? | `develop` |
| Description? | Upgrading the PHPDoc comments of this class, using the latest PS standards and also applying the PS standards to some of the code. Also getting the Franglais out of the comments, getting rid of things the Symfony code sniffer nags about, etc. It might have some additional `todo` tags, depending on whether the code was comprehensible or not. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | N/A |

Warning: this PR also has a code cleanup. Please check this first before merging!

### CODE REVIEW

File: classes/CMSCategory.php [hunk 1]
```diff
@@ -208,44 +239,60 @@ public static function getRecurseCategory($id_lang = null, $current = 1, $active
         return $category;
     }
 
-    public static function recurseCMSCategory($categories, $current, $id_cms_category = 1, $id_selected = 1, $is_html = 0)
+    /**
+     * @param     $categories
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]: $categories is an array, regarding the way we use it in this function.

File: classes/CMSCategory.php [hunk 2]
```diff
@@ -348,27 +399,34 @@ public static function getCategories($id_lang, $active = true, $order = true)
         foreach ($result as $row) {
             $categories[$row['id_parent']][$row['id_cms_category']]['infos'] = $row;
         }
+
         return $categories;
     }
 
-    public static function getSimpleCategories($id_lang)
+    /**
+     * @param $idLang
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]:
```diff
- * @param $idLang
+ * @param $idLang Language ID
```

File: classes/CMSCategory.php [hunk 3]
```diff
@@ -404,16 +463,24 @@ public static function hideCMSCategoryPosition($name)
...
+    public static function getHomeCategories($idLang, $active = true)
     {
-        return CMSCategory::getChildren(1, $id_lang, $active);
+        return CMSCategory::getChildren(1, $idLang, $active);
     }
 
-    public static function getChildren($id_parent, $id_lang, $active = true)
+    /**
+     * @param      $idParent
+     * @param      $idLang
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]:
```diff
- * @param $idLang
+ * @param $idLang Language ID
```

File: classes/CMSCategory.php [hunk 4]
```diff
@@ -468,26 +537,38 @@ public static function checkBeforeMove($id_cms_category, $id_parent)
         }
     }
 
-    public static function getLinkRewrite($id_cms_category, $id_lang)
+    /**
+     * @param $idCmsCategory
+     * @param $idLang
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]:
```diff
- * @param $idLang
+ * @param $idLang Language ID
```

File: classes/CMSCategory.php [hunk 5]
```diff
@@ -496,28 +577,34 @@ public function getLink(Link $link = null)
         return $link->getCMSCategoryLink($this->id, $this->link_rewrite);
     }
 
-    public function getName($id_lang = null)
+    /**
+     * @param null $idLang
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]: is it valid PHPDoc ?

```diff
- * @param null $idLang
+ * @param null|string|int $idLang Language ID
```

File: classes/CMSCategory.php [hunk 6]
```diff
@@ -496,28 +577,34 @@ public function getLink(Link $link = null)
...
+     *
+     * @return string
+     */
+    public function getName($idLang = null)
     {
         $context = Context::getContext();
-        if (!$id_lang) {
+        if (!$idLang) {
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]: can you replace it by ```if (!is_null($idLang)) {``` please ? less risky, more explicit, too.

File: classes/CMSCategory.php (hunk 1 above)
Comment (firstred) [inline classes/CMSCategory.php]: Needed that. Thanks!

File: classes/CMSCategory.php [hunk 7]
```diff
@@ -612,49 +685,66 @@ public function updatePosition($way, $position)
...
-			SET `position` = '.(int)$i.'
-			WHERE `id_parent` = '.(int)$id_category_parent.'
-			AND `id_cms_category` = '.(int)$result[$i]['id_cms_category'];
+			SET `position` = '.(int) $i.'
+			WHERE `id_parent` = '.(int) $idCategoryParent.'
+			AND `id_cms_category` = '.(int) $result[$i]['id_cms_category'];
             Db::getInstance()->execute($sql);
         }
+
         return true;
```
Comment (mickaelandrieu) [inline classes/CMSCategory.php]: The more I read this class, the more I want to improve it ^^

File: classes/CMSCategory.php (hunk 6 above)
Comment (firstred) [inline classes/CMSCategory.php]: Do you wan to invert it?

File: classes/CMSCategory.php (hunk 7 above)
Comment (firstred) [inline classes/CMSCategory.php]: Yeah, me too. But I don't want to make huge changes at once anymore. The first Cart PR was a total failure haha (expected `12`, got `0.0000`, errrr...).

File: classes/CMSCategory.php (hunk 6 above)
Comment (mickaelandrieu) [inline classes/CMSCategory.php]: not realy, just want to make it very explicit: for instance,

```php
if (!is_null($idLang)) { }
```
allows one and only one available value (null) to enter in the if condition.

Or maybe I didn't understood the previous condition, what do you think ?

Comment (firstred) [inline classes/CMSCategory.php]: Why `null`?
Anything that isn't a natural number isn't useful here.

Comment (mickaelandrieu) [inline classes/CMSCategory.php]: Yup, but I prefer see and fix the bug if any, than handle it the wrong way :)

Comment (firstred) [inline classes/CMSCategory.php]: Let's do that! You seem to know what you're doing.

Comment (firstred) [inline classes/CMSCategory.php]: Done

Comment (mickaelandrieu) [inline classes/CMSCategory.php]: Ok sorry, just got it my mistake :p 

```diff
- if (!is_null($idLang)) { }
+ if (is_null($idLang)) { }
```

Comment (firstred) [inline classes/CMSCategory.php]: Hmmz, I wanted to recommend `empty`, but now that we have 'contracts' in the PHPDoc comments, it's much better to go for your suggestion ;)

**Pull request closed without merging**